community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/StoreMigrator.java

```diff
@@ -170,7 +171,7 @@ public void migrate( File storeDir, File migrationDir, MigrationProgressMonitor.
 
         RecordFormats oldFormat = selectForVersion( versionToMigrateFrom );
         RecordFormats newFormat = selectForVersion( versionToMigrateTo );
-        if ( !oldFormat.hasSameCapabilities( newFormat, CapabilityType.FORMAT ) )
+        if ( upgradeFormatType( oldFormat, newFormat ) || isDifferentCapabilities( oldFormat, newFormat ) )
```

Comment:
This condition handle 2 cases - we have different families that we should migrate or we have same families but with different capabilities. Case when we have some not comparable families handled in `UpgradableDatabase`.
Maybe it's more naming thing that brings confusion since those 2 questions are related and highlight confusion around that?